File: user-service/src/main/java/rs/edu/raf/mapper/CompanyMapper.java
package rs.edu.raf.mapper;

import org.springframework.stereotype.Component;
import rs.edu.raf.dto.CompanyCreateDTO;
import rs.edu.raf.dto.CompanyDTO;
import rs.edu.raf.model.user.Company;

@Component
public class CompanyMapper {

    public Company companyCreateDTOtoCompany(CompanyCreateDTO companyCreateDTO) {
        Company company = new Company();

        company.setPhoneNumber(companyCreateDTO.phoneNumber());
        company.setAddress(companyCreateDTO.address());
        company.setName(companyCreateDTO.name());
        company.setTIN(companyCreateDTO.TIN());
        company.setBusinessActivityCode(companyCreateDTO.businessActivityCode());
        company.setFaxNumber(companyCreateDTO.faxNumber());
        company.setRegistrationNumber(companyCreateDTO.registrationNumber());
        company.setRegistryNumber(companyCreateDTO.registryNumber());

        return company;
    }

    public CompanyDTO companyToCompanyDTO(Company company) {
        return new CompanyDTO(company.getId(), company.getPhoneNumber(), company.getAddress(), company.getName(),
                company.getAccountNumbers(), company.getFaxNumber(), company.getTIN(), company.getRegistrationNumber(),
                company.getBusinessActivityCode(), company.getRegistryNumber());
    }
}

File: user-service-kotlin/src/main/kotlin/org/example/userservicekotlin/mapper/CompanyMapper.kt
package org.example.userservicekotlin.mapper

import org.example.userservicekotlin.dto.CompanyCreateDTO
import org.example.userservicekotlin.model.user.Company
import org.example.userservicekotlin.dto.CompanyDTO
import org.springframework.stereotype.Component

@Component
class CompanyMapper {
    fun companyCreateDTOtoCompany(companyCreateDTO: CompanyCreateDTO): Company {
        return Company(
            null,
            companyCreateDTO.phoneNumber,
            companyCreateDTO.address,
            companyCreateDTO.name,
            null,
            companyCreateDTO.faxNumber,
            companyCreateDTO.TIN,
            companyCreateDTO.registrationNumber,
            companyCreateDTO.businessActivityCode,
            companyCreateDTO.registryNumber
        )
    }

    fun companyToCompanyDTO(company: Company): CompanyDTO {
        return CompanyDTO(
            company.id,
            company.phoneNumber,
            company.address,
            company.name,
            company.accountNumbers,
            company.faxNumber,
            company.TIN,
            company.registrationNumber,
            company.businessActivityCode,
            company.registryNumber
        )
    }
}